routes/favicon.coffee

```diff
@@ -0,0 +1,28 @@
+request = require 'request'
+fs = require 'fs'
+cachefile = 'favicon.ico'
+
+favicon = (req, res, next) ->
+	try
+		res.send fs.readFileSync cachefile
+	catch error
+		request "https://vine.co/", (error, response, body) ->
```

Comment:
Using only async functions where possible actually was my initial approach, but then I was deterred by the deep indentation. The implementation is now back to async again.

The icon was always saved to the directory the program runs in, which is its base directory since 0036188, unless someone changed the `CACHEFILE` to an absolute path. This was the intention in this constant. `os.tmpdir` is a step into the right direction, but more correct would be some cache directory. [The cachedir package](https://github.com/linusu/node-cachedir) only does the right thing when run as user, under root it should for example direct to `/var/cache/vinefeed`. I am still searching for a package that can universally find such a location, but used `os.tmpdir` until something pops up.